modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpPipeliningHandler.java

```diff
@@ -109,11 +109,30 @@ public void write(final ChannelHandlerContext ctx, final Object msg, final Chann
         assert msg instanceof Netty4HttpResponse : "Invalid message type: " + msg.getClass();
         boolean success = false;
         try {
-            List<Tuple<Netty4HttpResponse, ChannelPromise>> readyResponses = write((Netty4HttpResponse) msg, promise);
-            for (Tuple<Netty4HttpResponse, ChannelPromise> readyResponse : readyResponses) {
-                doWrite(ctx, readyResponse.v1(), readyResponse.v2());
+            final Netty4HttpResponse response = (Netty4HttpResponse) msg;
+            if (response.getSequence() != writeSequence) {
+                if (outboundHoldingQueue.size() >= maxEventsHeld) {
+                    int eventCount = outboundHoldingQueue.size() + 1;
+                    throw new IllegalStateException(
```

Comment:
This PR slightly changes semantics for when the queue is nearly full in that if the event that triggers that the queue is full has the right seq-no, it no longer fails. That sounds fine to me, but wanted to ensure other reviewers consider it.